static/js/containers/ProfilePage.js

```diff
@@ -40,9 +39,7 @@ class ProfilePage extends React.Component {
       dispatch(startProfileEdit());
     }
     return dispatch(validateProfile(profile, requiredFields, messages)).then(() => {
-      dispatch(saveProfile(SETTINGS.username, profile)).then(() => {
-        dispatch(clearProfileEdit());
-      });
+      dispatch(saveProfile(SETTINGS.username, profile));
```

Comment:
We don't want to remove the `clearProfileEdit` dispatch here. We want to clear the edit object after we successfully save the edits.

the `clearProfileEdit` action here is only going to be dispatched after we've finished editing and we've successfully saved the object (well, after the `Promise` has been resolved), so it shouldn't affect the changes you're making.